### PR TITLE
Verify lineage for QueryOptionsFromQueryVisitor

### PR DESCRIPTION
**Pull request by lbschanno**
Write tests for QueryOptionsFromQueryVisitor to verify that it returns a
query with a valid lineage. Additionally, update the documentation, and
ensure that any AND/OR nodes with subsequently deleted children are
cleaned up.

Part of #880

### CODE REVIEW

**Comment by ivakegg** [comment]
This looks like a pull request for the unique query options thing, not the lineage thing.  Are we crossing the streams here?

**Comment by lbschanno** [comment]
@ivakegg the unique query options PR was already previously merged into integration. This focuses on just validating the overall lineage for QueryOptionsFromQueryVisitor.

**Review by apmoriarty — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryOptionsFromQueryVisitor.java [hunk 1]
```diff
@@ -49,14 +76,58 @@ public Object visit(ASTFunctionNode node, Object data) {
     }
     
     /**
-     * If this is the f:options function, descend the tree with a List in the passed userdata. The function args will be collected into the List when visiting
-     * the child ASTStringLiteral nodes
+     * If the passed OR node contains a f:options
```
Comment (jschmidt10) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryOptionsFromQueryVisitor.java]: Looks like part of this sentence may have been chopped off.

Comment (lbschanno) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryOptionsFromQueryVisitor.java]: Fixed!

**Review by ivakegg — APPROVED** [review]

**Pull request merged**